superset/db_engine_specs/ocient.py

```diff
@@ -114,3 +117,22 @@ def fetch_data(cls, cursor, lim=None):
         if len(data) != 0 and type(data[0]).__name__ == 'Row':
             data = [list(map(_verify_datatype, r)) for r in data]
         return data
+    
+    @classmethod
+    def get_cancel_query_id(cls, cursor: Any, query: Query) -> Optional[str]:
+        # Store the cursor locally so that we can retrieve it if we need to cancel
+        cursor_cache[query.id] = cursor
```

Comment:
This looks like a resource leak. How are entries removed from the map?

Reply:
Good point, let me check on that

Reply:
Entries are now removed upon cancellation or query completion